aleo-programs/staking_lite/src/main.leo

```diff
@@ -0,0 +1,402 @@
+      return 0u64;
+    }
+
+    let rewards_unsigned: u128 = rewards_since_last_claim as u128;
+    let new_commission: u128 = rewards_unsigned * commission_rate / PRECISION_UNSIGNED;
+    let new_commission_64: u64 = new_commission as u64;
+    return new_commission_64;
+  }
+
+  transition unbond_commission(amount: u64) {
+    assert_eq(self.caller, ADMIN);
+    credits.aleo/unbond_public(amount);
+
+    return then finalize(amount);
+  }
+
+  finalize unbond_commission(amount: u64) {
+    let current_state: u8 = state.get(0u8);
+    assert_eq(current_state, 0u8);
+
+    // Calculate commission earned since last claim
+    let bonded: u64 = 0u64;
+    let total_deposits: u64 = net_deposits.get(0u8);
+    let rewards_at_last_claim: u64 = last_commission_claim.get(0u8).remaining_rewards;
+    let commission_rate: u128 = commission_percent.get(0u8);
+    let commission_earned: u64 = calculate_commission_earned(bonded, total_deposits, rewards_at_last_claim, commission_rate);
+
+    assert(commission_earned > 0u64);
+    assert(commission_earned >= amount);
+
+    state.set(0u8, 1u8);
+
+    // Set last_commission_claim
+    let total_rewards: u64 = bonded - total_deposits;
+    let new_claim: commission_claim = commission_claim {
+      height: block.height + UNBONDING_PERIOD,
+      microcredits_claimed: commission_earned,
+      remaining_rewards: total_rewards - amount
+    };
+    last_commission_claim.set(0u8, new_claim);
+  }
+
+  transition claim_commission() {
+    credits.aleo/claim_unbond_public();
+
+    return then finalize();
+  }
+
+  finalize claim_commission() {
+    let current_state: u8 = state.get(0u8);
+    assert_eq(current_state, 1u8);
+
+    let unbonding_height: u32 = last_commission_claim.get(0u8).height;
+    assert(block.height >= unbonding_height);
+
+    // Update state
+    state.set(0u8, 0u8);
+  }
+
+  transition bond_all(validator_address: address, amount: u64) {
+    assert_eq(self.caller, ADMIN);
+    credits.aleo/bond_public(validator_address, amount);
+
+    return then finalize(validator_address, amount);
+  }
+
+  finalize bond_all(validator_address: address, amount: u64) {
+    let current_state: u8 = state.get(0u8);
+    assert_eq(current_state, 0u8);
+
+    let unbonding_balance: u64 = 0u64;
+    assert_eq(unbonding_balance, 0u64);
+
+    let account_balance: u64 = 0u64;
+    let pending_withdrawals: u64 = pending_withdrawal.get(0u8);
+    let available_balance: u64 = account_balance - pending_withdrawals;
+    assert(amount <= available_balance);
+  }
+
+  transition transfer_commission(recipient: address, amount: u64) {
+    assert_eq(self.caller, ADMIN);
+    credits.aleo/transfer_public(recipient, amount);
+
+    return then finalize(recipient, amount);
+  }
+
+  finalize transfer_commission(recipient: address, amount: u64) {
+    let current_state: u8 = state.get(0u8);
+    assert_eq(current_state, 0u8);
+
+    let account_balance: u64 = 0u64;
+    let unbonding_balance: u64 = 0u64;
```

Comment:
Are these placeholders for getting from the credits.aleo mapping? Otherwise, the below calculation is just 0+0